extensions/cms/src/cmsUtils.ts

```diff
@@ -70,59 +91,86 @@ export class CmsUtils {
 	}
 
 	public async getRegisteredServers(ownerUri: string, relativePath: string): Promise<mssql.ListRegisteredServersResult> {
-		return this.getCmsService().then((service) => {
-			return service.getRegisteredServers(ownerUri, relativePath).then((result) => {
-				if (result && result.registeredServersList && result.registeredServersList) {
-					return result;
-				}
-			});
-		});
+		const cmsService = await this.getCmsService();
+		const result = await cmsService.getRegisteredServers(ownerUri, relativePath);
+		if (result && result.registeredServersList && result.registeredServersList) {
+			return result;
+		}
 	}
 
 	public async createCmsServer(connection: azdata.connection.Connection,
-		name: string, description: string): Promise<mssql.ListRegisteredServersResult> {
+		name: string, description: string): Promise<CreateCmsResult> {
 		let provider = await this.getCmsService();
 		connection.providerName = connection.providerName === cmsProvider ? mssqlProvider : connection.providerName;
 		let ownerUri = await azdata.connection.getUriForConnection(connection.connectionId);
 		if (!ownerUri) {
 			// Make a connection if it's not already connected
-			await azdata.connection.connect(Utils.toConnectionProfile(connection), false, false).then(async (result) => {
-				ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
-			});
-		}
-		return provider.createCmsServer(name, description, connection, ownerUri).then((result) => {
-			if (result) {
-				return Promise.resolve(result);
-			} else {
-				return Promise.reject(null);
+			let initialConnectionProfile = this.getConnectionProfile(connection);
+			let result = await azdata.connection.connect(initialConnectionProfile, false, false);
+			ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
+			// If the ownerUri is still undefined, then open a connection dialog with the connection
+			if (!ownerUri) {
+				try {
+					let result = await this.connection(initialConnectionProfile);
+					if (result) {
+						ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
+						connection = result;
+					}
+				} catch (error) {
```

Comment:
Just get rid of the try catch if all you're doing is rethrowing (here and other places)